src/test/java/http/server/mocks/MockServerSocket.java

```diff
@@ -0,0 +1,26 @@
+package http.server.mocks;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.nio.charset.Charset;
+
+public class MockServerSocket extends ServerSocket {
```

Comment:
Not sure whether to have a mock extend the class it stands in for is good practice? I'm doing this to avoid having a bunch of interfaces and classes like `ServerSocketWrapper` solely for the purposes of testing.

Reply:
What are the pros and cons of extending? How about for wrapping?